src/node.rs

```diff
@@ -40,6 +41,8 @@ impl<C: TreeConfig> SyntaxNode<C> {
     }
 
     /// Returns an iterator over the direct child nodes and tokens of this node.
+    /// This *might* benefit from being a namable type.
+    /// Also, maybe the right API is `foo.children(&tree).tokens() / `foo.children(&tree).nodes()` ?
```

Comment:
I think it might even be possible to replace `child_nodes` and `child_tokens` with `children` + `filter_map`, though I’d think that would have an impact on performance. That aside, I prefer the `child_nodes`/`child_tokens` API, since I think it’s more obvious to the end-user.